samples/tutorials/hardwareInterfaces/i2c/BatteryChargerReading/batteryReadingComponent/batteryReading.c

```diff
@@ -0,0 +1,275 @@
+
+//-------------------------------------------------------------------------------------------------
+//* Static functions
+//--------------------------------------------------------------------------------------------------
+/**
+ * Get the file handle for the given I2C bus and configure it for access to the given I2C address.
+ *
+ * @return
+ *      - LE_FAULT on failure
+ *      - A filehandle to the I2C bus
+ */
+//--------------------------------------------------------------------------------------------------
+static int I2cAccessBusAddr
+(
+    uint8_t i2cBus,
+    uint8_t i2cAddr
+)
+{
+    const size_t filenameSize = 32;
+    char filename[filenameSize];
+
+    snprintf(filename, filenameSize, "/dev/i2c/%d", i2cBus);
+
+    LE_DEBUG("Opening I2C bus: '%s'", filename);
+    int fd = open(filename, O_RDWR);
+    if (fd < 0 && (errno == ENOENT || errno == ENOTDIR))
+    {
+        snprintf(filename, filenameSize, "/dev/i2c-%d", i2cBus);
+        LE_DEBUG("Opening I2C bus: '%s'", filename);
+        fd = open(filename, O_RDWR);
+    }
+
+    if (fd < 0)
+    {
+        if (errno == ENOENT)
+        {
+            LE_ERROR(
+                "Could not open file /dev/i2c-%d or /dev/i2c/%d: %s\n",
+                i2cBus,
+                i2cBus,
+                strerror(ENOENT));
+        }
+        else
+        {
+            LE_ERROR("Could not open file %s': %s\n", filename, strerror(errno));
+        }
+
+        return LE_FAULT;
+    }
+
+    if (ioctl(fd, I2C_SLAVE_FORCE, i2cAddr) < 0)
+    {
+        LE_ERROR("Could not set address to 0x%02x: %s\n", i2cAddr, strerror(errno));
+        return LE_FAULT;
+    }
+
+    return fd;
+}
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Performs an SMBUS read of a 1 byte register
+ *
+ * @return
+ *      - LE_OK
+ *      - LE_FAULT
+ */
+//--------------------------------------------------------------------------------------------------
+static le_result_t SmbusReadReg
+(
+    uint8_t i2cBus,  ///< [IN] I2C bus to perform the read on
+    uint8_t i2cAddr, ///< [IN] I2C address to read from
+    int8_t  reg,     ///< [IN] Register within the I2C device to read
+    uint8_t *data    ///< [OUT] Value stored within the register.  This value is only valid if the
+                     ///  function returned LE_OK.
+)
+{
+    int i2cFd = I2cAccessBusAddr(i2cBus, i2cAddr);
+    if (i2cFd == LE_FAULT)
+    {
+        LE_ERROR("failed to open i2c bus %d for access to address %d\n", i2cBus, i2cAddr);
+        return  LE_FAULT;
+    }
+
+    le_result_t result;
+
+    const int readResult = i2c_smbus_read_byte_data(i2cFd, reg);
+
+    if (readResult < 0)
+    {
+        LE_ERROR("smbus read failed with error %d", readResult);
+        result = LE_FAULT;
+    }
+    else
+    {
+        *data = readResult;
+        result = LE_OK;
+    }
+
+    close(i2cFd);
+
+    LE_DEBUG("SMBUS READ addr=0x%x, reg=0x%x, data=0x%x", i2cAddr, reg, *data);
```

Comment:
I think this `LE_DEBUG` statement belongs inside the `else` statement above so that it only gets printed when the read is successful.